umap/migrations/0018_datalayer_uuid.py

```diff
@@ -0,0 +1,42 @@
+# Generated by Django 4.2.8 on 2024-02-19 17:40
+
+import uuid
+
+from django.db import migrations, models
+
+
+def gen_uuid(apps, schema_editor):
+    DataLayer = apps.get_model("umap", "DataLayer")
+    for row in DataLayer.objects.all():
+        row.uuid = uuid.uuid4()
+        row.save(update_fields=["uuid"])
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("umap", "0017_migrate_to_openstreetmap_oauth2"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="datalayer",
+            name="uuid",
+            field=models.UUIDField(default=uuid.uuid4, editable=False, null=True),
+        ),
+        migrations.RunPython(gen_uuid, reverse_code=migrations.RunPython.noop),
+        migrations.RunSQL(
+            "ALTER TABLE umap_datalayer DROP CONSTRAINT umap_datalayer_pkey"
```

Comment:
I wonder if this will work on other systems, as the constraints seems to be named uniquely on different installs.